app/components/hooks/useGetChainIdsToDetectNfts.test.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  DeepPartial,
+  renderHookWithProvider,
+} from '../../util/test/renderWithProvider';
+import { backgroundState } from '../../util/test/initial-root-state';
+import { RootState } from '../../reducers';
+import { useGetChainIdsToDetectNfts } from './useGetChainIdsToDetectNfts';
+import { useSelector } from 'react-redux';
+import {
+  selectChainId,
+  selectIsAllNetworks,
+  selectIsPopularNetwork,
+  selectAllPopularNetworkConfigurations,
+} from '../../selectors/networkController';
+
+const mockInitialState: DeepPartial<RootState> = {
+  settings: {},
+  engine: {
+    backgroundState: {
+      ...backgroundState,
+    },
+  },
+};
+
+jest.mock('react-redux', () => ({
+  ...jest.requireActual('react-redux'),
+  useSelector: jest.fn(),
+}));
+
+describe('useGetChainIdsToDetectNfts', () => {
+  it('should return current chain id when isAllNetworks is false', async () => {
```

Comment:
S: "isAllNetworks is false" is the implementation, explain the goal of this test instead. When is isAllNetworks false ? Is this when token filter is showing all networks? If so, isAllNetworks is false means only the current network?
If that's right, try:
```suggestion
  it('returns current chain id when filtered on current network', async () => {
```
this allows others to understand the meaning, the goal of you test.  The implementation and how to get this info is in the test itself and it's selector mock.